pubsublite/spark-connector/spark_streaming_to_pubsublite_example.py

```diff
@@ -35,13 +36,32 @@ def spark_streaming_to_pubsublite(
     # |-- value: long (nullable = true)
     sdf = spark.readStream.format("rate").option("rowsPerSecond", 1).load()
 
+    # Transform the dataframe to match the required data fields and data types:
+    # https://github.com/googleapis/java-pubsublite-spark#data-schema
     sdf = (
-        sdf.withColumn("key", (sdf.value % 5).cast(StringType()).cast(BinaryType()))
-        .withColumn("event_timestamp", sdf.timestamp)
-        .withColumn("data", sdf.value.cast(StringType()).cast(BinaryType()))
-        .drop("value", "timestamp")
+        sdf.withColumn("key", lit("example").cast(BinaryType()))
+        .withColumn("data", col("value").cast(StringType()).cast(BinaryType()))
+        .withColumnRenamed("timestamp", "event_timestamp")
+        # Populate the attributes field. For example, an even value will
+        # have {"key1", [b"even"]}.
+        .withColumn(
+            "attributes",
+            create_map(
+                lit("key1"),
+                array(when(col("value") % 2 == 0, b"even").otherwise(b"odd")),
+            ),
+        )
+        .drop("value")
     )
 
+    # After the transformation, the schema of the dataframe should look like:
+    # |-- key: binary (nullable = false)
+    # |-- data: binary (nullable = true)
+    # |-- event_timestamp: timestamp (nullable = true)
+    # |-- attributes: map (nullable = false)
```

Comment:
our guide and everywhere actually specifies attributes is nullable.

Reply:
but our code doesn't really check since they are compatible, I think it's fine.

Reply:
@jiangmichaellll  Sounds like we need to update our guide? When you say "everywhere", which places do you refer to?